messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/ExternalJMSQueueService.java

```diff
@@ -36,19 +53,94 @@
+            try {
+                QueueConnectionFactory cf = (QueueConnectionFactory) storeBaseContext.lookup(pcfInjector.getValue().getBindInfo().getAbsoluteJndiName());
+                try (QueueConnection connection = cf.createQueueConnection()) {
+                    QueueSession session = connection.createQueueSession(false, Session.AUTO_ACKNOWLEDGE);
+                    connection.start();
+                    QueueRequestor requestor = new QueueRequestor(session, managementQueue);
+                    Message m = session.createMessage();
+                    if (config.getSelector() != null && !config.getSelector().isEmpty()) {
+                        org.apache.activemq.artemis.api.jms.management.JMSManagementHelper.putOperationInvocation(m, ResourceNames.BROKER, "createQueue", JMS_QUEUE_PREFIX + queueName, JMS_QUEUE_PREFIX + queueName, config.getSelector(), config.isDurable(), RoutingType.ANYCAST.name());
+                    } else {
+                        org.apache.activemq.artemis.api.jms.management.JMSManagementHelper.putOperationInvocation(m, ResourceNames.BROKER, "createQueue", JMS_QUEUE_PREFIX + queueName, JMS_QUEUE_PREFIX + queueName, config.isDurable(), RoutingType.ANYCAST.name());
+                    }
+                    Message reply = requestor.request(m);
+                    ROOT_LOGGER.debugf("Creating queue %s returned %s", JMS_QUEUE_PREFIX + queueName, reply);
+                    boolean success = reply.getBooleanProperty("_AMQ_OperationSucceeded");
+                    if(success) {
+                        ROOT_LOGGER.infof("Queue %s has been created", JMS_QUEUE_PREFIX + queueName);
+                    } else {
+                        ROOT_LOGGER.errorf("Creating queue %s failed with error %s", JMS_QUEUE_PREFIX + queueName, reply.getBody(String.class));
+                    }
+                }
+            } catch (NamingException | JMSException ex) {
+                    throw new StartException(ex);
+            } finally {
+                try {
+                    storeBaseContext.close();
+                } catch (NamingException ex) {
+                    throw new StartException(ex);
+                }
+            }
+        }
         queue = ActiveMQJMSClient.createQueue(JMS_QUEUE_PREFIX + queueName);
     }
 
     @Override
     public synchronized void stop(final StopContext context) {
+        NamingStore namingStore = namingStoreInjector.getOptionalValue();
+        if(namingStore!= null) {
+            Queue managementQueue = ActiveMQJMSClient.createQueue("activemq.management");
+            final NamingContext storeBaseContext = new NamingContext(namingStore, null);
+            try {
+                QueueConnectionFactory cf = (QueueConnectionFactory) storeBaseContext.lookup(pcfInjector.getValue().getBindInfo().getAbsoluteJndiName());
+                try (QueueConnection connection = cf.createQueueConnection()) {
```

Comment:
what if the remote Artemis has enabled authentication on the management address?

Reply:
Good point